aten/src/ATen/native/xnnpack/OpContext.cpp

```diff
@@ -0,0 +1,59 @@
+#ifdef USE_XNNPACK
+#include <ATen/native/xnnpack/Convolution.h>
+#include <ATen/native/xnnpack/Linear.h>
+#include <ATen/native/xnnpack/OpContext.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+
+c10::intrusive_ptr<XNNPackLinearOpContext>
+XNNPackLinearOpContext::create_context(at::Tensor&& weight,
+    c10::optional<at::Tensor>&& bias,
+    const c10::optional<double> output_min,
+    const c10::optional<double> output_max) {
+  auto linear_op_context =
+	  c10::make_intrusive<XNNPackLinearOpContext>(std::move(weight), std::move(bias));
+  linear_op_context->op_context_ = xnnpack::internal::linear::create(
+        linear_op_context->orig_weight_,
+        linear_op_context->orig_bias_,
+        output_min ? *output_min : xnnpack::Context::kMin,
+        output_max ? *output_max : xnnpack::Context::kMax);
+  return linear_op_context;
+}
+
+c10::intrusive_ptr<XNNPackConv2dOpContext>
+XNNPackConv2dOpContext::create_context(at::Tensor&& weight,
+    c10::optional<at::Tensor>&& bias,
+    std::vector<int64_t>&& padding,
+    std::vector<int64_t>&& stride,
+    std::vector<int64_t>&& dilation,
```

Comment:
Likewise all here and other places in code potentially with the exception of Operator which is just a non-copyable std::unique_ptr.  I would personally just opt for option 1 or 2 above 95%+ of the time.  Even 3 is overkill unless one is writing a generic library.

Reply:
As I mentioned before, passing by value maybe little expensive, but I suppose create_context is called only once at load time so it might be ok. If you this seems more clean, I am ok with it, but as I said, I wanted to explain my reasoning and see if I might be wrong in that. Then something new to learn :).